Main styling

laurenyz opened the pull request:
## Description
- updated input fields to material-ui components
- added top navbar
- added a copy modal which copies the share code to the clipboard 
- added logout functionality to the navbar which returns the user to the homepage
- adjusted clear icon button
- remove unused styles leftover from `create-react-app`

<!-- What does this code change? Why did I choose this approach? Did I learn anything worth sharing? Reminder: This will be a publicly facing representation of your work (READ: help you land that sweet dev gig). -->
We wanted the styling of this app to be straightforward and modern.  We wanted to add more functionality to enable users to access multiple lists by logging out and to make it easier for users to share list tokens.

## Related Issue

<!-- If you write "closes" followed by the Github issue number, it will automatically close the issue for you when the PR merges -->

## Acceptance Criteria

<!-- Include AC from the Github issue -->

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |
|  ✓ | :sparkles: Update styling     |

## Updates

### Before
![image](https://user-images.githubusercontent.com/50968795/94217102-2c0cf500-feaf-11ea-8a3e-a59fa7dacb7d.png)
![image](https://user-images.githubusercontent.com/50968795/94217113-33340300-feaf-11ea-90fe-afdffe5b91ce.png)

### After
![image](https://user-images.githubusercontent.com/50968795/94217124-38914d80-feaf-11ea-881b-daa44baf6f07.png)
![image](https://user-images.githubusercontent.com/50968795/94217135-3dee9800-feaf-11ea-95c4-045184d19ff5.png)
![image](https://user-images.githubusercontent.com/50968795/94217141-42b34c00-feaf-11ea-8994-8f0be13d44e9.png)


## Testing Steps / QA Criteria

- Create a new list or use an existing token
- Make sure that the clear button still works when searching for an item.
- Check that you can copy the share token by clicking on the share icon in the navbar and clicking the `copy` button.
- Check that you can log out by clicking the logout icon in the navbar and that you are automatically directed back to the homepage.
- Check that the share and logout icons only appear once you've entered a share token and disappear when you exit a list.

QMaximillian committed:
Replace base inputs with Material UI TextField components

QMaximillian committed:
Add header component with Typography Material UI component to  AddItem.js to match 'Smart Shopppping List' header text in List.js

QMaximillian committed:
Merge branch 'main-styling' into ql-input-styling

QMaximillian committed:
Add Material UI Button styling

Add base styling for AddItem.js, 'Add To List' button, List.js button that erases the input, and Details.js back button

QMaximillian committed:
Remove 'X' and replace with Material UI 'CloseIcon'

QMaximillian committed:
Add Typography component for text inside of AddItem.js 'Add To Item' button component

laurenyz committed:
adjusted search input to conditionally show clear button

QMaximillian committed:
Remove padding from ClearIcon to prevent InputAdornment jumping motion when changing between icons

QMaximillian committed:
Remove separated clear input button for input in List.js

Input styling PR will include a way to clear the input if the input is not empty and show a SearchIcon if it is empty. This logic will fully hide the CloseIcon Material UI icon.

QMaximillian committed:
Add button styling for Welcome.js

Style 'Create New List' Link as a Button. Style 'Submit' button

QMaximillian committed:
Add TextField input for Welcome.js 'Enter Token' input, create a component InputProps endAdornment key to clear an input

QMaximillian committed:
Update index.html, manifest.json, and add Helmet components to AddItem.js, Details.js, List.js, and Welcome.js

Remove create-react-app favicons and add a crystal ball emoji as an svg favicon. Remove title  tag from index.html in order to dynamically update page titles with React Helmet. Update manifest.json to list name and shortname as CartVoyant, our apps name. Give each component listed above a title and meta tag inside Helmet tag. Give App.js an additional link tag for the canonical url

QMaximillian committed:
Merge pull request #55 from the-collab-lab/ql-button-styling

Ql button styling

QMaximillian committed:
Fix merge conflicts

QMaximillian committed:
Merge pull request #52 from the-collab-lab/ql-input-styling

Ql input styling

TianaJohnson committed:
Nav Bar CSS

 Css pushing nav bar to the bottom of the page and making it consistant across all pages

QMaximillian committed:
Remove unused styling

Remove *.module.css files. Remove unused styles from App.css. Remove inline styling from List.js and put styling in App.css. Delete header tag from App.js. Add simple styling to keep List.js ul list in place.

laurenyz committed:
added logo, title and brief description

laurenyz committed:
added nav bar with logo, title

added copy and logout buttons that render conditionally if there is a token

added copy functions, which opens a modal on click, and a logout function that returns to the homepage

TianaJohnson committed:
Fixing up bottom navigation

TianaJohnson committed:
Merge branch 'main-styling' of https://github.com/the-collab-lab/tcl-10-smart-shopping-list into bottom-nav-finetune-tj

laurenyz committed:
removed autofocus from copymodal

laurenyz committed:
Merge pull request #59 from the-collab-lab/ly-navbar

Ly navbar

QMaximillian committed:
Merge main-styling

TianaJohnson committed:
Merge branch 'main-styling' of https://github.com/the-collab-lab/tcl-10-smart-shopping-list into bottom-nav-finetune-tj

QMaximillian committed:
Merge pull request #57 from the-collab-lab/ql-remove-unused-styling

Remove unused styling

QMaximillian committed:
Merge ql-add-react-helmet-and-semantic-page-tags to combine head tags across all pages

Preserve current index.html head tag adjustments including title and logo while adding React Helmet tags for individual routes to have their own title and meta tags

laurenyz committed:
updated custom theme colors and removed black background from app.css. Background currently reading from material_ui_theme.js

QMaximillian committed:
Merge branch 'main-styling' into ly-adding-title-and-logo

laurenyz committed:
updated material_ui_theme

TianaJohnson committed:
Finishing Touches

 Tab coloring

TianaJohnson committed:
Edits

 Removed old css

 Adjusted root makeStyles

laurenmbeatty commented on src/components/CopyModal.js:
```diff
@@ -0,0 +1,44 @@
+          type="submit"
+          onClick={copy}
+          startIcon={<AssignmentIcon />}
+        >
```
I would give the button an aria-label that with `copy token` or something like that.

laurenyz replied:
Sounds good!

laurenmbeatty commented on src/components/CopyModal.js:
```diff
@@ -0,0 +1,44 @@
+  );
+};
+
+export default CopyModal;
```
Love the idea of this modal!

TianaJohnson committed:
Merge branch 'main-styling' into bottom-nav-finetune-tj

laurenmbeatty commented on src/components/TopNav.js:
```diff
@@ -0,0 +1,85 @@
+  );
+}
+
+export default TopNav;
```
A good accessibility practice is to either use text with an icon, or to at least have an `aria-label` associated with them.

laurenmbeatty commented on src/components/Welcome.js:
```diff
@@ -30,20 +32,34 @@ const Welcome = ({ setToken }) => {
           <label htmlFor="join-list">Enter a share token</label>
         </div>
-        <input
+        <TextField
```
If you want the Button and TextField to be more in alignment, wrap them in a div and set:
`display: flex, justifyContent: center`
![image](https://user-images.githubusercontent.com/10378331/94262784-9f3b5900-fef9-11ea-8a5f-bf4c55466261.png)

laurenmbeatty commented:
Maybe make this button the same style as the others (not as rounded):
![image](https://user-images.githubusercontent.com/10378331/94262880-cd209d80-fef9-11ea-8b4d-b8689e97ef83.png)

QMaximillian committed:
Remove unused styling in App.css

QMaximillian committed:
Remove crystal ball svg

QMaximillian committed:
Merge branch 'main-styling' into ly-adding-title-and-logo

QMaximillian committed:
Adjust styling to prevent AppBar from changing size based on changing from Welcome view to List view

TianaJohnson committed:
Edits and updates

 Added a custom divider

TianaJohnson committed:
Merge branch 'bottom-nav-finetune-tj' of https://github.com/the-collab-lab/tcl-10-smart-shopping-list into bottom-nav-finetune-tj

TianaJohnson committed:
Fixed merge conflicts

QMaximillian committed:
Merge pull request #58 from the-collab-lab/ly-adding-title-and-logo

Ly ql adding logo and semantic head tags to index.html and each route page

QMaximillian committed:
Merge pull request #62 from the-collab-lab/ql-fix-appbar-styling

Ql fix appbar styling

QMaximillian committed:
Adjust styles to fit all app content within viewport

Use Material UI makeStyles to layyout app styles for app container, the main content of the app, and each navbar. Add Box components to place each section positionally

laurenyz committed:
added aria labels for icon buttons

laurenyz committed:
added aria label for copy button

laurenyz committed:
removed button custom button styling to make it less round as per suggestion

added spacing to input and button

added list token text to add item view

laurenyz committed:
adjusted sizing and spacing of text when list is empty

laurenmbeatty commented on src/components/TopNav.js:
```diff
@@ -0,0 +1,87 @@
+                  <Grid item>
+                    <IconButton
+                      onClick={() => setOpenCopy(true)}
+                      aria-label={`copy list token`}
```
👍

laurenmbeatty commented on src/components/CopyModal.js:
```diff
@@ -0,0 +1,45 @@
+          type="submit"
+          onClick={copy}
+          startIcon={<AssignmentIcon />}
+          aria-label={`copy list token: ${token}`}
```
💯

QMaximillian committed:
Merge branch 'main-styling' into ql-main-content-flex-container

QMaximillian committed:
Merge pull request #63 from the-collab-lab/ql-main-content-flex-container

Adjust styles to fit all app content within viewport

QMaximillian committed:
Adjust BottomNav styling

Remove extra styling, adjust Tabs component indicator color

QMaximillian committed:
Fix merge conflicts

QMaximillian committed:
Fix extra line

QMaximillian committed:
Add semantic naming for tab state

QMaximillian committed:
Merge pull request #61 from the-collab-lab/bottom-nav-finetune-tj

Bottom Navigation Styling

QMaximillian committed:
Fix tab number setting error

laurenmbeatty commented:
Looks great on mobile! if you give the footer
`position: absolute,
 bottom: 0,
 left: 0,
 right: 0`

It will always be affixed to the bottom in non-mobile, as well. Sometimes on the list page it gets lost. 
![image](https://user-images.githubusercontent.com/10378331/94354284-62797a00-003f-11eb-9eef-9a2d0e38c626.png)

laurenyz committed:
updated success alert to material-ui snackbar with an auto hide duration of 3 seconds

QMaximillian committed:
Welcome screen styling

QMaximillian committed:
Merge branch 'main-styling' of github.com:the-collab-lab/tcl-10-smart-shopping-list into main-styling

laurenyz committed:
updated details modal to material-ui dialog so accessible by keyboard

laurenyz committed:
Merge branch 'main-styling' of https://github.com/the-collab-lab/tcl-10-smart-shopping-list into main-styling

laurenyz committed:
moved delete confirm message to modal

laurenyz committed:
added aria labels for snackbar alerts

QMaximillian committed:
Attempt to layout grid stretching across entire container

QMaximillian committed:
Merge branch 'main-styling' of github.com:the-collab-lab/tcl-10-smart-shopping-list into main-styling

laurenyz committed:
increased size in delete modal message

laurenyz committed:
adjusted spacing in list item when list is empty

QMaximillian committed:
Adjust height of List.js ul children li elements. Add CircularProgress component to show during App loading state

QMaximillian committed:
Convert Add Item NavLink in List.js to a Button that functions as a NavLink

laurenyz committed:
added empty bottom nav for welcome page, adjusted demo sizing for main app container and for list

laurenyz committed:
adjusted marginTop for demo

laurenyz committed:
moved snackbar alerts to top right and set maxWidth for delete modal for demo

laurenyz committed:
added maxWidth for demo

QMaximillian committed:
Style fixes

QMaximillian committed:
Merge branch 'main-styling' of github.com:the-collab-lab/tcl-10-smart-shopping-list into main-styling

laurenyz committed:
changed copy modal button color to purple

laurenmbeatty commented on src/assets/material_ui_theme.js:
```diff
@@ -3,14 +3,15 @@ import { createMuiTheme } from '@material-ui/core/styles';
     secondary: {
-      main: '#eceef7',
+      main: '#551a8b', //dark purple
+      //main: '#aa8dd8', light purple
```
Do you still need this?

laurenmbeatty commented on src/App.js:
```diff
@@ -37,55 +61,79 @@ function App() {
-              }
+    <Box className="App">
+      <Grid container justify="center">
+        {/* <Grid item xs={12} sm={6} md={4} style={{height: "90vh", marginTop: "5vh"}}> more responsive */}
```
Are you saving this for after the demo? Or can delete?

QMaximillian replied:
Saving it for after the demo!

laurenmbeatty commented on src/components/AddItem.js:
```diff
@@ -59,78 +67,149 @@ const AddItem = ({ token, results }) => {
+        message={`Successfully added ${name}to your list.`}
+        action={<CheckBoxIcon />}
+        role="alert"
+        aria-label={`Successfully added ${name}to your list.`}
```
Nice

laurenmbeatty commented on src/components/AddItem.js:
```diff
@@ -59,78 +67,149 @@ const AddItem = ({ token, results }) => {
-          value={name}
-          onChange={e => setName(e.target.value)}
-          id="add-shopping-list-item"
+    <React.Fragment>
```
You can also just use `<>` and `</>` for fragments now.

laurenmbeatty commented on src/components/List.js:
```diff
@@ -1,22 +1,37 @@
+
+const useStyles = makeStyles({
+  box: {
+    height: '48vh',
```
This style falls apart a little bit depending on how tall the browser window is:
![image](https://user-images.githubusercontent.com/10378331/94364850-ee1bf680-0091-11eb-9bc8-e1399dceeb27.png)

laurenmbeatty reviewed (approved):
WOWWWWW!!!!! Gorgeous gorgeous gorgeous!  

I love how you really explored and took full advantage of MaterialUI's API. And Helmet...so cool.

laurenmbeatty commented:
One of the great things about using an already-created framework (one that does a good job, anyway) is the built-in accessibility. This is completely keyboard-navigable. Notice how when you open the details or delete modal, the focus transfers to the modal, and then when the modal closes, it transfers the focus back to the previous focused element.

QMaximillian committed:
Adjust flex styling for AddItem.js

QMaximillian committed:
Styling changes for Details modal

QMaximillian committed:
Add temporary vertical spacing in DeleteModal
